### PR TITLE
start mapping new enchantment system

### PR DESCRIPTION
**Pull request by eristhea**

### CODE REVIEW

File: mappings/net/minecraft/enchantment/EnchantmentHelper.mapping [hunk 1]
```diff
@@ -1,4 +1,66 @@
...
+	METHOD m_kvkboaps forEachEquippedEnchantment (Lnet/minecraft/unmapped/C_usxaxydn;Lnet/minecraft/unmapped/C_jakrppis$C_wtxpyhvg;)V
+		ARG 0 entity
+		ARG 1 enchantmentConsumer
+	METHOD m_lkbuzftr getProjectileSpeed (Lnet/minecraft/unmapped/C_bdwnwhiu;Lnet/minecraft/unmapped/C_sddaxwyk;Lnet/minecraft/unmapped/C_astfners;F)F
+		ARG 2 entity
+		ARG 3 baseSpeed
+	METHOD m_mcehpxsn getBlockExperience (Lnet/minecraft/unmapped/C_bdwnwhiu;Lnet/minecraft/unmapped/C_sddaxwyk;I)I
+		ARG 2 baseBlockExperience
+	METHOD m_mimuzvcl supportsEnchantments (Lnet/minecraft/unmapped/C_sddaxwyk;)Z
+	METHOD m_obijpcqw getRandomItemWith (Lnet/minecraft/unmapped/C_pscqxfcs;Lnet/minecraft/unmapped/C_usxaxydn;Ljava/util/function/Predicate;)Ljava/util/Optional;
```
Comment (supersaiyansubtlety) [inline mappings/net/minecraft/enchantment/EnchantmentHelper.mapping]:
```suggestion
	METHOD m_obijpcqw getContextForRandomEquipmentWith (Lnet/minecraft/unmapped/C_pscqxfcs;Lnet/minecraft/unmapped/C_usxaxydn;Ljava/util/function/Predicate;)Ljava/util/Optional;
```
verbose, I know
but `getRandomItemWith` makes it sound like an `Item` is returned, and doesn't mention equipment

File: mappings/net/minecraft/enchantment/EnchantmentHelper.mapping [hunk 2]
```diff
@@ -1,4 +1,66 @@
...
+		ARG 1 enchantmentConsumer
+	METHOD m_anckqezf tick (Lnet/minecraft/unmapped/C_bdwnwhiu;Lnet/minecraft/unmapped/C_usxaxydn;)V
+	METHOD m_aqtkrfxn applyLocationBasedEffects (Lnet/minecraft/unmapped/C_bdwnwhiu;Lnet/minecraft/unmapped/C_sddaxwyk;Lnet/minecraft/unmapped/C_usxaxydn;Lnet/minecraft/unmapped/C_yuycoehb;)V
+	METHOD m_byjosehc getLevel (Lnet/minecraft/unmapped/C_cjzoxshv;Lnet/minecraft/unmapped/C_sddaxwyk;)I
+		ARG 0 enchantment
+	METHOD m_chdruiod getDurabilityChange (Lnet/minecraft/unmapped/C_bdwnwhiu;Lnet/minecraft/unmapped/C_sddaxwyk;I)I
+		ARG 2 baseDamage
+	METHOD m_cjzxcjvw applyAttributeModifiers (Lnet/minecraft/unmapped/C_sddaxwyk;Lnet/minecraft/unmapped/C_qfoqahef;Ljava/util/function/BiConsumer;)V
+		ARG 1 group
+		ARG 2 attributeModifierConsumer
```
Comment (supersaiyansubtlety) [inline mappings/net/minecraft/enchantment/EnchantmentHelper.mapping]:
```suggestion
		ARG 2 applicator
```
?

File: mappings/net/minecraft/enchantment/EnchantmentHelper.mapping [hunk 3]
```diff
@@ -15,10 +77,78 @@ CLASS net/minecraft/unmapped/C_jakrppis net/minecraft/enchantment/EnchantmentHel
...
 		COMMENT
 		COMMENT <p>For enchanted books, it sets the enchantments to the item stack's
 		COMMENT stored enchantments than regular enchantments.
 		ARG 0 stack
+	METHOD m_yolmdyfp onProjectileSpawned (Lnet/minecraft/unmapped/C_bdwnwhiu;Lnet/minecraft/unmapped/C_sddaxwyk;Lnet/minecraft/unmapped/C_gstiksvs;Ljava/util/function/Consumer;)V
+		ARG 2 projectile
+		ARG 3 onBreak
+	METHOD m_zkmvjzhi applyAttributeModifiers (Lnet/minecraft/unmapped/C_sddaxwyk;Lnet/minecraft/unmapped/C_yuycoehb;Ljava/util/function/BiConsumer;)V
+		ARG 1 slot
+		ARG 2 attributeModifierConsumer
```
Comment (supersaiyansubtlety) [inline mappings/net/minecraft/enchantment/EnchantmentHelper.mapping]:
```suggestion
		ARG 2 applicator
```
?

File: mappings/net/minecraft/enchantment/effect/AllOf.mapping
Comment (supersaiyansubtlety) [inline mappings/net/minecraft/enchantment/effect/AllOf.mapping:1]: I'd prefer something closer to loom's `AllOfEnchantmentEffects`.
`AllOf` begs the questions "What's an `AllOf`?", "All of what?"

**Review by supersaiyansubtlety — CHANGES_REQUESTED** [review]

**Review by OroArmor — APPROVED** [review]
I think the enchantment provider classes `C_dpsarzgt`, `C_swckftoh`, and `C_ycmdmexf` need mapping along with the providers classes: p (EnchantmentProviders): `C_fudougzs` and `C_hzfgfhbe`. This could be a seperate PR thought

File: mappings/net/minecraft/enchantment/EnchantmentHelper.mapping (hunk 1 above)
Comment (eristhea) [reply mappings/net/minecraft/enchantment/EnchantmentHelper.mapping]: fair, i can't really think of a better way to communicate everything

File: mappings/net/minecraft/enchantment/effect/AllOf.mapping
Comment (eristhea) [reply mappings/net/minecraft/enchantment/effect/AllOf.mapping:1]: to me the package answers quite a bit of that, i'm ok with `AllOfEffects` though

**Comment by eristhea** [comment]
> I think the enchantment provider classes `C_dpsarzgt`, `C_swckftoh`, and `C_ycmdmexf` need mapping along with the providers classes: p (EnchantmentProviders): `C_fudougzs` and `C_hzfgfhbe`. This could be a seperate PR thought

I'll map those in my next PR

**Review by supersaiyansubtlety — APPROVED** [review]

**Pull request merged**